src/client/client/networking.js

```diff
@@ -6,17 +6,24 @@ import * as gameConstants from "../../game-constants.json";
 import { marbleManager } from "../marble-manager";
 import { Vector3 } from "three";
 import * as msgPack from "msgpack-lite";
+import { updateManager } from "../update-manager";
 
-let networking = function() {
-	let _wsUri = `ws${config.ssl ? "s" : ""}://${window.location.hostname}${config.websockets.localReroute ? "" : `:${config.websockets.port}`}/ws/gameplay`;
+const networking = function() {
+	const _wsUri = `ws${config.ssl ? "s" : ""}://${window.location.hostname}${config.websockets.localReroute ? "" : `:${config.websockets.port}`}/ws/gameplay`;
 	let _ws = null;
 
 	let _updateBuffer = []; // Array of game updates, each containing events and marble data
 	let _timeDeltaRemainder = null; // Represents the timeDelta between two updates, or null if there are none
 	let _desiredBufferSize = config.defaultBufferSize; // Desired buffer size
 	let _previousMarblePositions = null;
 
-	let _processMessageEvent = function(event) {
+	// Page visibility logic. If the page is hidden, the network buffer will fast-forward when a race ends
+	// https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState
+	let isPageHidden = false;
+	document.addEventListener("visibilitychange", () => {isPageHidden = document.visibilityState === "hidden";}, false);
+	window.addEventListener("pagehide", () => {isPageHidden = document.visibilityState === "hidden";}, false); // Safari fix, which doesn't trigger the visibilitychange event
```

Comment:
These two callback functions are exactly the same. It might be nicer to store the function in a variable and use that twice, but I'll leave that up to you.